addons/mrp/models/mrp_production.py

```diff
@@ -332,6 +332,7 @@ def _compute_state(self):
                 production.state = 'draft'
             elif all(move.state == 'cancel' for move in production.move_raw_ids):
                 production.state = 'cancel'
+                production.action_cancel()
```

Comment:
Don't call the action_cancel in the compute. Override button_cancel in purchase_mrp